### ISSUE TITLE
[Epic] ML Integration — cuML Wrappers and PyTorch/TorchGeo Interop

### ISSUE THREAD

**Issue by jayavenkatesh19** [jayavenkatesh19/terragpu#5]
## Overview
Simple ML for scientists who don't need deep learning, plus clean handoff to PyTorch/TorchGeo for those who do.

### cuML Wrappers (`tg.ml`)
- `tg.ml.classify(stack, labels, method="random_forest")` — supervised classification
- `tg.ml.cluster(stack, method="kmeans", n_clusters=5)` — unsupervised clustering
- Methods: random_forest, xgboost, kmeans, dbscan (all via cuML)
- Automatically flattens RasterStack to tabular format for cuML, reshapes predictions back to raster
- Train/predict split helpers

### PyTorch Interop
- `stack.to_torch()` — convert to GPU tensor, zero-copy where possible via DLPack
- `stack.to_torch(dtype=torch.float32, device="cuda:0")` — explicit control
- Preserves geospatial metadata as tensor attributes or companion dict

### TorchGeo Interop
- `stack & labels` — dataset composition operator (inspired by TorchGeo's & operator)
- Returns a TorchGeo-compatible GeoDataset object
- Compatible with TorchGeo's RandomGeoSampler and DataLoader workflow
- Dictionary-based returns: `batch["image"]`, `batch["mask"]`

### Other Interop
- `stack.to_xarray()` — xarray Dataset with geo metadata
- `stack.to_numpy()` / `stack.to_cupy()` — raw array access
- `stack.to_dataframe()` — cuDF/pandas tabular representation

## Design Reference
```python
# Simple ML (no PyTorch needed)
model = tg.ml.classify(stack, labels, method="random_forest")
prediction = model.predict(stack)

# PyTorch handoff
X = stack.to_torch()

# TorchGeo handoff
dataset = stack & labels
sampler = RandomGeoSampler(dataset, size=256, length=1000)
loader = DataLoader(dataset, sampler=sampler, batch_size=16)
```

## Boundary
TerraGPU does NOT replicate TorchGeo's training infrastructure (trainers, datamodules, pretrained models). The handoff should be clean and complete.
